services/catalog/src/simcore_service_catalog/services/director.py

```diff
@@ -136,6 +140,23 @@ async def is_responsive(self) -> bool:
         except (httpx.HTTPStatusError, httpx.RequestError, httpx.TimeoutException):
             return False
 
+    async def list_all_services(self) -> list[ServiceMetaDataPublished]:
+        # WARNING: this function probably raise ValidationError since director does NOT offer guarantees.
+        # SEE list_registered_services
+        data = await self.get("/services")
+        return parse_obj_as(list[ServiceMetaDataPublished], data)
+
```

Comment:
what you write here is correct. so why do you not do something like
```python
validated_metadata = await asyncio.gather(asyncio.get_event_loop().run_in_executor(None, parse_obj_as(ServiceMetaDataPublished)), return_exception=True)
return [_ for _ in validated_metadata if not isinstance(_, BaseException)]
```

Reply:
@sanderegg I see the intention here but I am a bit confused about the details. You propose to convert the validation into an async function but how do you want to gather it? i.e. one executor per item of the list or partioned? Should this be a job for the new gather function you created?

Reply:
If you end up going for such a function, please limit gather concurrency somehow